### PR TITLE
Add maven release info, usage and badge to README.

### PR DESCRIPTION
**Pull request by dblock**
*Issue #, if available:*

*Description of changes:*

Screen Shot 2022-07-13 at 12 42 43 PM

### Pull Request Checklist:

- [x] I have added a contribution line at the top of [CHANGELOG.md](CHANGELOG.md).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +11,16 @@ This library is based on [AWS Interceptor](https://github.com/awslabs/aws-reques
 
 ## Usage
 
+Add [io.github.acm19.aws-request-signing-apache-interceptor](https://repo1.maven.org/maven2/io/github/acm19/aws-request-signing-apache-interceptor/) as a dependency.
+
+```xml
+<dependency>
+  <groupId>io.github.acm19</groupId>
+  <artifactId>aws-request-signing-apache-interceptor</artifactId>
+  <version>2.1.1</version>
```
Comment (acm19) [inline README.md:20]: It's important that we update this as part of the release process, in `update_release_files_and_commit` recipe for example. A simple `sed` would do the trick.

Comment (dblock) [reply README.md:20]: I knew you were going to say that. Updated. Also fixed CHANGELOG getting a -SNAPSHOT version and when you try to release it leaves -SNAPSHOT in there.

Comment (acm19) [reply README.md:20]: :) That was kind of intentional.

File: Makefile [hunk 2]
```diff
@@ -46,9 +48,10 @@ set_next_development_version:
 .PHONY: update_development_files_and_commit
 .SILENT: update_development_files_and_commit
 update_development_files_and_commit:
-	sed -i "1 s/^/### $(shell mvn help:evaluate -Dexpression=project.version -q -DforceStdout) \(Next\)\\n\\n/" CHANGELOG.md
-	git add pom.xml CHANGELOG.md
-	git commit -m "Prepare for next development iteration $(shell mvn help:evaluate -Dexpression=project.version -q -DforceStdout)"
+	VERSION=$(shell mvn help:evaluate -Dexpression=project.version -q -DforceStdout | cut -d- -f1) && \
+	sed -i "1 s/^/### $$VERSION \(Next\)\\n\\n/" CHANGELOG.md && \
+	git add pom.xml CHANGELOG.md && \
+	git commit -m "Prepare for next development iteration $$VERSION"
```
Comment (acm19) [inline Makefile:54]: I'd leave the the SNAPSHOT here at least.

**Review by acm19 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (acm19) [reply README.md:20]: I think it's great now btw, I'd only leave the snapshot on the next working version comment, but I don't feel strong about it, so feel free to merge it.

Comment (dblock) [reply README.md:20]: Yeah, I kinda prefer not to have SNAPSHOT in it because the actual version we plan to develop is not -SNAPSHOT.

**Pull request merged**